guide: hotfix pipelines links

jorgeorpinel opened the pull request:
Closes #3945

See for example link to "Avoiding unexpected behavior" at the end of https://dvc-org-guide-pipelines-6zpmjd.herokuapp.com/doc/user-guide/project-structure/internal-files (review app).

> Note that no redirect is needed because we never published path `/doc/user-guide/data-pipelines`, it was just an internal mistake in https://github.com/iterative/dvc.org/pull/3414/files.

jorgeorpinel commented:
☝🏼 All links are passing, just need approve+merge from someone please.

jorgeorpinel commented:
p.s. I wonder why the link checker didn't catch all the broken links in https://github.com/iterative/dvc.org/pull/3414/files 🤷🏼

dtrifiro commented:
The `avoiding-unexpected-behavior` section does not exist on the linked page though (https://dvc-org-guide-pipelines-6zpmjd.herokuapp.com/doc/user-guide/pipelines/defining-pipelines#avoiding-unexpected-behavior)

jorgeorpinel commented:
Hey yeah @dtrifiro great double catch then! I realized that I removed that (left for an upcoming page about Pipeline repro) but didn't remove the links. I put it back in https://dvc-org-guide-pipelines-6zpmjd.herokuapp.com/doc/user-guide/project-structure/dvcyaml-files#stage-commands for now (and updated the links). It's deploying, give it a few min.

jorgeorpinel commented:
Note that the checks are failing due to a [Prettier bug](https://github.com/iterative/dvc.org/pull/3947#pullrequestreview-1107937905).

jorgeorpinel commented:
Ready: https://dvc-org-guide-pipelines-6zpmjd.herokuapp.com/doc/user-guide/project-structure/dvcyaml-files#avoiding-unexpected-behavior

rogermparent reviewed (approved):
I don't see any problems from a webdev perspective!

rogermparent commented:
> p.s. I wonder why the link checker didn't catch all the broken links in https://github.com/iterative/dvc.org/pull/3414/files 🤷🏼

I think the checker still has issues with the alternate link style, though I remember we added a change to try and support it. The wonderful world of regex-parsing for links :tada:
